src/core/cluster.py

```diff
@@ -462,22 +482,31 @@ def _balancer_status(self) -> Status:
         if not self.peer_cluster_relation and not self.runs_broker:
             return Status.NO_PEER_CLUSTER_RELATION
 
-        if not self.balancer.broker_connected:
+        if not self.peer_cluster.broker_connected:
             return Status.NO_BROKER_DATA
 
-        if len(self.balancer.broker_capacities.get("brokerCapacities", [])) < MIN_REPLICAS:
+        if len(self.peer_cluster.broker_capacities.get("brokerCapacities", [])) < MIN_REPLICAS:
             return Status.NOT_ENOUGH_BROKERS
 
         return Status.ACTIVE
 
     @property
-    def _broker_status(self) -> Status:
+    def _broker_status(self) -> Status:  # noqa: C901
         """Checks for role=broker specific readiness."""
         if not self.runs_broker:
             return Status.ACTIVE
 
-        # FIXME
-        if not self.runs_controller:
+        # Neither ZooKeeper or KRaft are active
+        if self.kraft_mode is None:
+            return Status.MISSING_MODE
+
+        if self.kraft_mode:
+            if not self.peer_cluster.controller_quorum_uris:  # FIXME: peer_cluster or cluster?
+                return Status.NO_QUORUM_URIS
+            if not self.cluster.cluster_uuid:
+                return Status.NO_CLUSTER_UUID
+
+        if self.kraft_mode == False:  # noqa: E712
```

Comment:
**suggestion**: Actually, I do not have an issue at all with `kraft_mode` being a `bool | None`. We could keep this [three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic) instead of raising in `self.kraft_mode`.

We would keep the `is None` check on L500, use `is False` here and could get rid of the E712 ignore instruction.
That being said, I understand the rationale of raising instead of using `None` because it's more error-prone, as we do not have a good way to enforce people to _not_ use `if not self.kraft_mode`